### DISCUSSION TITLE
Requirements with DocAsCode

### DISCUSSION THREAD

**Discussion by TomFleischmann** [CSA-FEDERATE/Proposed-BuildingBlocks#2] [Ideas]
SDV-OEM view in the area of detailed software requirements 

Several model based approaches for systems engineering and methods and tools are already established and state-of-the-art and available on the market, as well established in other industries (avionics, defense, etc.). The challenge in the automotive sector lies strongly in the variety of variants. 

Driven by different product variants and the possibility of choice by the customer, this results in a high degree of combinatorics and almost exponential complexity in the subsystems. 

In contrast to other industries, the automotive market with high international competition and price pressure, a wide variety of variants, functional safety and security with very long product life cycles of the vehicles in the field and the necessary software updates and hardware upgrades come together. It has become apparent that most complex, tool-based approaches to managing software requirements cannot keep pace with the capabilities of modern software development through git with the inherent capabilities of code management and automation. 

If the source code of the systems is maintained in variants (branches) and developed by means of CI/CD (Continuous Integration & Development) up to the control unit, the software requirements must also be manageable and versionable, just like the source code. This then allows the transfer of customized requirements from one branch to another (merge), the comparison of requirements (diff) and the automated check that all requirements along the necessary V-model artifacts (architecture, design, code, tests, test results) are covered (tracing). 

To this end, we as OEMs see it as essential to use these so-called "DocAsCode" approaches more efficiently directly in the development of ECU software internally as well as along the supply chain. Of course, this does not exempt us from overall systems engineering and model-based development, in any case we consider basic research to be less relevant there.

Proposal to add to the wiki for requirement toolchain: [Open Fast Trace](https://github.com/itsallcode/openfasttrace )

**Comment by prillerp** [comment]
We will add a new BB
BB-EST / Requirements / BB_DocAsCode.md (and import Tom's suggestions)
In addition we will add these subfolders to BB-EST:
BB-EST / Design
BB-EST / Implementation
BB-EST / BuildAndIntegration
BB-EST / Testing
BB-EST / Deployment
BB-EST / MonitoringAndDiagnostics
BB-EST / LifecycleManagement
